docs/source/getting_started/Installing.rst

```diff
@@ -17,25 +17,157 @@ with many of packages required by *msticpy* pre-installed.
 Creating a virtual environment
 ------------------------------
 
+.. note:: This is an optional step. You will most likely want to do this
+   if you are installing msticpy in a local Python installation. If
+   you are using a cloud notebook environment such as Azure ML you
+   will usually not need to create a virtual environment.
+
 *msticpy* has a significant number of dependencies. To avoid conflicts
 with packages in your existing Python environment you may want to
 create a Python virtual environment
 or a conda environment and install the package there.
 
 For standard python use the ``venv`` command to do this
-(there are several alternatives to venv available).
-For Conda use the conda ``env`` command.
-In both cases be sure to activate the environment before running jupyter
-using ``activate {my_env_name}`` (or ``conda activate {my_env_name}``).
+(there are also several alternatives to venv available).
+
+.. code:: bash
+
+    ~$ python -m venv my_env
+    ~$ ./my_env/scripts/activate
+    (my_env) ~$
+
+For Conda use the conda ``create`` command from a conda shell.
+
+.. code:: bash
+
+    (base) c:\users\ian> conda create -n my_env
+    (base) c:\users\ian> conda activate my_env
+    (my_env) c:\users\ian>
+
+You should see the name of the environment that you've just
+created and activated in the prompt.
+
 
 Installation
 ------------
 
-Run the following command to install *msticpy*.
+Run the following command to install the base configuation of *msticpy*.
 
 
 ``pip install msticpy``
 
 or for the latest dev build
 
-``pip install git+https://github.com/microsoft/msticpy``
\ No newline at end of file
+``pip install git+https://github.com/microsoft/msticpy``
+
+
+Selective Installation - using "extras"
+---------------------------------------
+
+pip supports specification of an additional parameter sequence
+known as extras. The syntax for this is:
+
+``pip install package_name[extra1, extra2...]``
+
+As of version 0.9.0 *msticpy* has its dependencies split into
+extras. This allows you to install only the packages that you
+need and avoid the overhead of time and diskspace of dependencies
+that you do not need.
+
+.. note:: extras do not affect the which code from *msticpy* is
+   installed - only the external libraries on which certain
+   functions inside *msticpy* need to work.
+
+Extras in msticpy
+~~~~~~~~~~~~~~~~~
+
+The extras available in *msticpy* are described in the following table:
+
++------------------+------------------------------------+--------------+--------------+
+| extra            | Functionality                      | Install time | Install time |
+|                  |                                    | (increment)  | (full)       |
++==================+====================================+==============+==============+
+| default install  | - Most functionality (approx 75%)  |       --     |   1m:13s     |
```

Comment:
I think we should explicitly call out that this won't install KQL stuff. Given KQL is the core of the notebooks we have produced so far I think its worth highlighting this.

Reply:
I know you say this below but if other people are like me they will just jump straight to the table.

Reply:
I've put a big warning before this table